extensions/wikia/BrokenRenameFix/SpecialBrokenRenameFixController.class.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Class SpecialBrokenRenameFixController
+ *
+ * A special page with an interface to rerun local rename jobs.
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ * @copyright (c) 2015 Wikia, Inc.
+ * @license http://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
+ * @package BrokenRenameFix
+ */
+
+class SpecialBrokenRenameFixController extends WikiaSpecialPageController {
+
+	const BRF_FORM_SUBMIT = 'submit';
+
+	private
+		$notices = [],
+		$userId,
+		$oldName,
+		$newName;
+
+	function __construct() {
+		parent::__construct( 'BrokenRenameFix', 'renameuser', true );
+	}
+
+	public function index() {
+		Wikia::addAssetsToOutput( 'special_broken_rename_fix' );
+
+		$this->wg->Out->setPageTitle( wfMessage( 'brf-title' )->escaped() );
```

Comment:
There's no permissions check here right now, so an anon or any non-staff member could access this form.